site/en/blog/immutable-document-domain/index.md

```diff
@@ -0,0 +1,197 @@
+
+On `https://parent.example.com`:
+
+```js
+// Send a message to https://video.example.com
+iframe.postMessage('Request DOM manipulation', 'https://video.example.com');
+
+// Receive messages
+iframe.addEventListener('message', (event) => {
+  // Reject all messages except ones from https://video.example.com
+  if (event.origin !== 'https://video.example.com') return;
+
+  // Filter success messages
+  if (event.data === 'succeeded') {
+    // DOM manipulation is succeeded
+  }
+});
+```
+
+On `https://video.example.com`:
+
+```js
+// Receive messages
+window.addEventListener('message', (event) => {
+  // Reject all messages except ones from https://parent.example.com
+  if (event.origin !== 'https://parent.example.com') return;
+
+  // Do a DOM manipulation on https://video.example.com.
+
+  // Send a success message to https://parent.example.com
+  event.source.postMessage('succeeded', event.origin);
+});
+```
+
+Try it and see if it works. If you have specific requirements that won't work
+with `postMessage()` or Channel Messaging API, please let us know on [Twitter
+via @ChromiumDev](https://twitter.com/ChromiumDev) or ask on Stack Overflow.
+
+### Send `Origin-Agent-Cluster: ?0` header to continue using `document.domain`
+
+If you have strong reasons to continue setting `document.domain`, you can do so
+by sending `Origin-Agent-Cluster: ?0` response header along with the target
+document.
+
+```http
+Origin-Agent-Cluster: ?0
+```
+
+`Origin-Agent-Cluster` header instructs the browser whether the document should
+be handled by the origin-keyed agent cluster or not. To learn more about Origin
+Agent Cluster, read [Requesting performance isolation with the
+Origin-Agent-Cluster header](https://web.dev/origin-agent-cluster/).
+
+By sending this header, the document can continue setting `document.domain` even
+after it becomes immutable by default.
+
+## Timeline
+
+Chrome will eventually make `document.domain` immutable. Here's the current
+timeline:
+
+* **Chrome 98:** Usage of `document.domain` displays a warning message in
+  DevTools Console. Also, `Origin-Agent-Cluster` header is available.
+* **Chrome 99:** Enterprise policy is in place to extend behavior.
+* **Chrome 101:** `document.domain` is immutable.
+
+## Browser compatibility
+
+* [In the Origin
+  spec](https://html.spec.whatwg.org/multipage/origin.html#:~:text=Because%20of%20these%20security%20pitfalls%2C%20this%20feature%20is%20in%20the%20process%20of%20being%20removed%20from%20the%20web%20platform),
+  it's clearly stated that the feature should be removed.
+* [WebKit indicated that they are moderately positive about deprecating
+  `document.domain`
+  setter](https://github.com/w3ctag/design-reviews/issues/564#issuecomment-768450217).
+
+## Resources
+
+* [Document.domain - Web APIs |
+  MDN](https://developer.mozilla.org/docs/Web/API/Document/domain)
+* [mikewest/deprecating-document-domain: `document.domain` intentionally weakens
```

Comment:
Is this supposed to be in the doc? If yes, definitely edit the wording to be more clear and less casual.